daemon/gamemode.c

```diff
@@ -59,6 +59,11 @@ POSSIBILITY OF SUCH DAMAGE.
  */
 #define NICE_DEFAULT_PRIORITY -4
 
+/* Little helper to safely print into a buffer, returns a newly allocated string
+ */
+#define safe_snprintf(b, s, ...)                                                                   \
+	(snprintf(b, sizeof(b), s, __VA_ARGS__) < (ssize_t)sizeof(b) ? strndup(b, sizeof(b)) : NULL)
+
```

Comment:
Don't see why this needs to allocate a new string as well seeing as we've already supplied the buffer? Could just return b or null as an indicator of success.

Reply:
Also, this commit is in the wrong order - "Add function to resolve the wine preloader executable" is before it and depends on it.

Reply:
The idea was having a reuseable and disposable buffer explicitely, and having a string as a result I can work with without having to think of `strdup()`'ing it extra. I actually had it the way you suggested it first and it was a complete mess. I had multiple buffers per stack or had to `strdup()` all the time just to make sure I would not overwrite my string in the next buffer re-use. There's maybe one code location now where we do a `strdup()` too much now but I don't think it's worth microoptimizing this. If you want to re-use the helper in other parts of the code, I could think of making two versions: One with and one without the `strdup()`. But currently it makes my code more readable and shorter.

Order: Yes, I noticed that, too. I checked in the functions one after each other during a longer edit session, then rebased afterwards to put it into correct commit order. Github insists on displaying this in order of commit date, instead of parent commit order (and even that gets messed up with partial rebases). Use `tig` on the command line to see it correctly. :-)

Reply:
PS: In compare view, is actually in correct order - https://github.com/FeralInteractive/gamemode/compare/master...kakra:identify-wine-exe

I think it's in a different order here for better overview of what changed after a rebase, to easy up incremental reviews.

Reply:
When I do the final code cleanup, I will look into making two versions of the helper and see if it optimizes things. But I think the helper is fine in view of how it is supposed to be used. Do you agree? A non-`strdup()` version clearly makes sense where the buffer is a one-time use only. I'll look into that but otoh I hope that you agree with me that a `strdup()` version is worth having. :-)

Reply:
@aejsmith I've added a non-`strdup()` version of the helper. But when used improperly, it may introduce more headache than it optimizes... I'm not sure if we want to keep it in. You decide...

Reply:
I understand your answer below as "yes, keep it in" and resolve this conversation. :-)